pkg/builder/builder.go

```diff
@@ -2327,6 +2331,15 @@ func GetTarballName(cmd []string) string {
 	return ""
 }
 
+// GetPasswordSecret returns the password secret
+func GetPasswordSecret(vdb *vapi.VerticaDB) string {
+	if vdb.Spec.PasswordSecret == vdb.Status.PasswordSecret {
+		return vdb.Spec.PasswordSecret
+	}
+	// status holds the current password
+	return vdb.Status.PasswordSecret
```

Comment:
It should be, if status is non-empty, return status. Otherwise return spec.

Reply:
This should always be vdb.Status.PasswordSecret, or adding passwordSecret after the DB created would fail with authentication error.